src/main/java/interaction/rule/BonusTankInteractionRule.java

```diff
@@ -19,9 +22,13 @@
 
 public class BonusTankInteractionRule implements InteractionRule {
 
+    private final BonusObjectManager bonusObjectManager;
+    private final TankObjectManager tankObjectManager;
     private final GameObjectManager gameObjectManager;
 
     public BonusTankInteractionRule(GameObjectManager gameObjectManager) {
+        this.bonusObjectManager = gameObjectManager.getBonusObjectManager();
+        this.tankObjectManager = gameObjectManager.getTankObjectManager();
         this.gameObjectManager = gameObjectManager;
```

Comment:
we can use either bonusObjectManager and tankObjectManager or gameObjectManager. Please consider only one of them.